### PR TITLE
r/sagemaker_domain - new resource

### PR DESCRIPTION
**Pull request by DrFaust92**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Relates #14453

Release note for [CHANGELOG](https://github.com/hashicorp/terraform-provider-aws/blob/master/CHANGELOG.md):

```release-note
resource_aws_sagemaker_domain - add new resource
resource_aws_sagemaker_image - fix error on wait for delete when image does not exist
```

Output from acceptance testing:

```
$ make testacc TESTARGS='-run=TestAccAWSSagemakerDomain_'
--- PASS: TestAccAWSSagemakerDomain_basic (228.59s)--- PASS: TestAccAWSSagemakerDomain_kernelGatewayAppSettings (233.93s)
--- PASS: TestAccAWSSagemakerDomain_disappears (222.39s)
--- PASS: TestAccAWSSagemakerDomain_jupyterServerAppSettings (223.22s)
--- PASS: TestAccAWSSagemakerDomain_sharingSettings (233.47s)
--- PASS: TestAccAWSSagemakerDomain_tensorboardAppSettings (237.31s)
--- PASS: TestAccAWSSagemakerDomain_securityGroup (288.18s)
--- PASS: TestAccAWSSagemakerDomain_tags (346.14s)
--- PASS: TestAccAWSSagemakerDomain_kms (242.96s)
```

### CODE REVIEW

File: website/docs/r/sagemaker_domain.html.markdown [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `domain_name` - (Required) The domain name.
+* `auth_mode` - (Required) The mode of authentication that members use to access the domain. Valid values are `IAM` and `SSO`.
+* `vpc_id` - (Required) The ID of the Amazon Virtual Private Cloud (VPC) that Studio uses for communication.
+* `subnet_ids` - (Required) The VPC subnets that Studio uses for communication.
+* `default_user_settings` - (Required) The default user settings. see [Default User Settings](#default-user-settings) below.
```
Comment (ewbankkit) [inline website/docs/r/sagemaker_domain.html.markdown:55]:
```suggestion
* `default_user_settings` - (Required) The default user settings. See [Default User Settings](#default-user-settings) below.
```

File: website/docs/r/sagemaker_domain.html.markdown [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+* `tags` - (Optional) A map of tags to assign to the resource.
+
+### Default User Settings
+
+* `execution_role` - (Required) The execution role ARN for the user.
+* `security_groups` - (Optional) The security groups.
+* `sharing_settings` - (Optional) The sharing settings. see [Sharing Settings](#sharing-settings) below.
+* `tensor_board_app_settings` - (Optional) The TensorBoard app settings. see [TensorBoard App Settings](#tensorboard-app-settings) below.
+* `jupyter_server_app_settings` - (Optional) The kernel gateway app settings. see [Jupyter Server App Settings](#jupyter-server-app-settings) below.
+* `kernel_gateway_app_settings` - (Optional) The Jupyter server's app settings. see [Kernel Gateway App Settings](#kernal-gateway-app-settings) below.
```
Comment (ewbankkit) [inline website/docs/r/sagemaker_domain.html.markdown:66]:
```suggestion
* `jupyter_server_app_settings` - (Optional) The Jupyter server's app settings. See [Jupyter Server App Settings](#jupyter-server-app-settings) below.
* `kernel_gateway_app_settings` - (Optional) The kernel gateway app settings. See [Kernel Gateway App Settings](#kernal-gateway-app-settings) below.
```

File: website/docs/r/sagemaker_domain.html.markdown [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+* `subnet_ids` - (Required) The VPC subnets that Studio uses for communication.
+* `default_user_settings` - (Required) The default user settings. see [Default User Settings](#default-user-settings) below.
+* `app_network_access_type` - (Optional) Specifies the VPC used for non-EFS traffic. The default value is `PublicInternetOnly`. Valid values are `PublicInternetOnly` and `VpcOnly`.
+* `tags` - (Optional) A map of tags to assign to the resource.
+
+### Default User Settings
+
+* `execution_role` - (Required) The execution role ARN for the user.
+* `security_groups` - (Optional) The security groups.
+* `sharing_settings` - (Optional) The sharing settings. see [Sharing Settings](#sharing-settings) below.
```
Comment (ewbankkit) [inline website/docs/r/sagemaker_domain.html.markdown:63]:
```suggestion
* `sharing_settings` - (Optional) The sharing settings. See [Sharing Settings](#sharing-settings) below.
```

File: website/docs/r/sagemaker_domain.html.markdown [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+* `tensor_board_app_settings` - (Optional) The TensorBoard app settings. see [TensorBoard App Settings](#tensorboard-app-settings) below.
```
Comment (ewbankkit) [inline website/docs/r/sagemaker_domain.html.markdown:64]:
```suggestion
* `tensor_board_app_settings` - (Optional) The TensorBoard app settings. See [TensorBoard App Settings](#tensorboard-app-settings) below.
```

File: website/docs/r/sagemaker_domain.html.markdown [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+* `s3_output_path` - (Optional) When `notebook_output_option` is Allowed, the Amazon S3 bucket used to save the notebook cell output.
+
+#### TensorBoard App Settings
+
+* `default_resource_spec` - (Optional) The default instance type and the Amazon Resource Name (ARN) of the SageMaker image created on the instance. see [Default Resource Spec](#default-resource-spec) below.
+
+#### Kernel Gateway App Settings
+
+* `default_resource_spec` - (Optional) The default instance type and the Amazon Resource Name (ARN) of the SageMaker image created on the instance. see [Default Resource Spec](#default-resource-spec) below.
+
```
Comment (ewbankkit) [inline website/docs/r/sagemaker_domain.html.markdown:81]: `KernelGatewayAppSettings` also has a `CustomImages` member: https://docs.aws.amazon.com/sagemaker/latest/APIReference/API_KernelGatewayAppSettings.html

File: aws/resource_aws_sagemaker_domain.go [hunk 6]
```diff
@@ -0,0 +1,596 @@
...
+	domainArn := aws.StringValue(output.DomainArn)
+	domainID, err := decodeSagemakerDomainID(domainArn)
+	if err != nil {
+		return err
+	}
+
+	d.SetId(domainID)
+
+	if _, err := waiter.DomainInService(conn, d.Id()); err != nil {
+		return fmt.Errorf("error waiting for sagemaker domain (%s) to create: %w", d.Id(), err)
```
Comment (ewbankkit) [inline aws/resource_aws_sagemaker_domain.go:248]:
```suggestion
		return fmt.Errorf("error waiting for SageMaker domain (%s) to create: %w", d.Id(), err)
```

File: aws/resource_aws_sagemaker_domain.go [hunk 7]
```diff
@@ -0,0 +1,596 @@
...
+							},
+						},
+						"kernel_gateway_app_settings": {
+							Type:     schema.TypeList,
+							Optional: true,
+							ForceNew: true,
+							MaxItems: 1,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"default_resource_spec": {
```
Comment (ewbankkit) [inline aws/resource_aws_sagemaker_domain.go:175]: `KernelGatewayAppSettings` also has a `CustomImages` member: https://docs.aws.amazon.com/sagemaker/latest/APIReference/API_KernelGatewayAppSettings.html

File: aws/resource_aws_sagemaker_domain.go [hunk 8]
```diff
@@ -0,0 +1,596 @@
...
+	d.Set("auth_mode", domain.AuthMode)
+	d.Set("app_network_access_type", domain.AppNetworkAccessType)
+	d.Set("arn", arn)
+	d.Set("home_efs_file_system_id", domain.HomeEfsFileSystemId)
+	d.Set("single_sign_on_managed_application_instance_id", domain.SingleSignOnManagedApplicationInstanceId)
+	d.Set("url", domain.Url)
+	d.Set("vpc_id", domain.VpcId)
+
+	if err := d.Set("subnet_ids", flattenStringSet(domain.SubnetIds)); err != nil {
+		return fmt.Errorf("error setting subnet_ids for sagemaker domain (%s): %w", d.Id(), err)
```
Comment (ewbankkit) [inline aws/resource_aws_sagemaker_domain.go:279]:
```suggestion
		return fmt.Errorf("error setting subnet_ids for SageMaker domain (%s): %w", d.Id(), err)
```

File: aws/resource_aws_sagemaker_domain.go [hunk 9]
```diff
@@ -0,0 +1,596 @@
...
+	}
+
+	if err := d.Set("default_user_settings", flattenSagemakerDomainDefaultUserSettings(domain.DefaultUserSettings)); err != nil {
+		return fmt.Errorf("error setting default_user_settings for sagemaker domain (%s): %w", d.Id(), err)
```
Comment (ewbankkit) [inline aws/resource_aws_sagemaker_domain.go:283]:
```suggestion
		return fmt.Errorf("error setting default_user_settings for SageMaker domain (%s): %w", d.Id(), err)
```

File: aws/resource_aws_sagemaker_domain.go [hunk 10]
```diff
@@ -0,0 +1,596 @@
...
+	}
+
+	tags, err := keyvaluetags.SagemakerListTags(conn, arn)
+
+	if err != nil {
+		return fmt.Errorf("error listing tags for Sagemaker Domain (%s): %w", d.Id(), err)
```
Comment (ewbankkit) [inline aws/resource_aws_sagemaker_domain.go:289]:
```suggestion
		return fmt.Errorf("error listing tags for SageMaker Domain (%s): %w", d.Id(), err)
```

File: aws/resource_aws_sagemaker_domain.go [hunk 11]
```diff
@@ -0,0 +1,596 @@
...
+		if err != nil {
+			return fmt.Errorf("error updating SageMaker domain: %w", err)
+		}
+	}
+
+	if d.HasChange("tags") {
+		o, n := d.GetChange("tags")
+
+		if err := keyvaluetags.SagemakerUpdateTags(conn, d.Get("arn").(string), o, n); err != nil {
+			return fmt.Errorf("error updating Sagemaker Notebook Instance (%s) tags: %s", d.Id(), err)
```
Comment (ewbankkit) [inline aws/resource_aws_sagemaker_domain.go:319]:
```suggestion
			return fmt.Errorf("error updating SageMaker domain (%s) tags: %w", d.Id(), err)
```

File: aws/resource_aws_sagemaker_domain.go [hunk 12]
```diff
@@ -0,0 +1,596 @@
...
+
+	if _, err := conn.DeleteDomain(input); err != nil {
+		if !isAWSErr(err, sagemaker.ErrCodeResourceNotFound, "") {
+			return fmt.Errorf("error deleting SageMaker domain (%s): %w", d.Id(), err)
+		}
+	}
+
+	if _, err := waiter.DomainDeleted(conn, d.Id()); err != nil {
+		if !isAWSErr(err, sagemaker.ErrCodeResourceNotFound, "") {
+			return fmt.Errorf("error waiting for sagemaker domain (%s) to delete: %w", d.Id(), err)
```
Comment (ewbankkit) [inline aws/resource_aws_sagemaker_domain.go:341]:
```suggestion
			return fmt.Errorf("error waiting for SageMaker domain (%s) to delete: %w", d.Id(), err)
```

**Comment by ewbankkit** [comment]
Verified acceptance tests:

```console
$ make testacc TEST=./aws/ TESTARGS='-run=TestAccAWSSagemakerDomain_' ACCTEST_PARALLELISM=1
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -count 1 -parallel 1 -run=TestAccAWSSagemakerDomain_ -timeout 120m
=== RUN   TestAccAWSSagemakerDomain_basic
=== PAUSE TestAccAWSSagemakerDomain_basic
=== RUN   TestAccAWSSagemakerDomain_tags
=== PAUSE TestAccAWSSagemakerDomain_tags
=== RUN   TestAccAWSSagemakerDomain_securityGroup
=== PAUSE TestAccAWSSagemakerDomain_securityGroup
=== RUN   TestAccAWSSagemakerDomain_sharingSettings
=== PAUSE TestAccAWSSagemakerDomain_sharingSettings
=== RUN   TestAccAWSSagemakerDomain_tensorboardAppSettings
=== PAUSE TestAccAWSSagemakerDomain_tensorboardAppSettings
=== RUN   TestAccAWSSagemakerDomain_kernelGatewayAppSettings
=== PAUSE TestAccAWSSagemakerDomain_kernelGatewayAppSettings
=== RUN   TestAccAWSSagemakerDomain_jupyterServerAppSettings
=== PAUSE TestAccAWSSagemakerDomain_jupyterServerAppSettings
=== RUN   TestAccAWSSagemakerDomain_disappears
=== PAUSE TestAccAWSSagemakerDomain_disappears
=== CONT  TestAccAWSSagemakerDomain_basic
--- PASS: TestAccAWSSagemakerDomain_basic (168.39s)
=== CONT  TestAccAWSSagemakerDomain_kernelGatewayAppSettings
--- PASS: TestAccAWSSagemakerDomain_kernelGatewayAppSettings (185.72s)
=== CONT  TestAccAWSSagemakerDomain_disappears
--- PASS: TestAccAWSSagemakerDomain_disappears (269.66s)
=== CONT  TestAccAWSSagemakerDomain_jupyterServerAppSettings
--- PASS: TestAccAWSSagemakerDomain_jupyterServerAppSettings (242.89s)
=== CONT  TestAccAWSSagemakerDomain_sharingSettings
=== CONT  TestAccAWSSagemakerDomain_tensorboardAppSettings
--- PASS: TestAccAWSSagemakerDomain_sharingSettings (185.68s)
=== CONT  TestAccAWSSagemakerDomain_securityGroup
--- PASS: TestAccAWSSagemakerDomain_tensorboardAppSettings (165.46s)
--- PASS: TestAccAWSSagemakerDomain_securityGroup (195.05s)
=== CONT  TestAccAWSSagemakerDomain_tags
--- PASS: TestAccAWSSagemakerDomain_tags (280.55s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	1693.469s
```

Parallelism set to 1 or else I get errors like

```
Error: error creating SageMaker domain: ResourceLimitExceeded: The account-level service limit 'Total domains' is 1 Domains, with current utilization of 1 Domains and a request delta of 1 Domains. Please contact AWS support to request an increase for this limit.
```

File: aws/resource_aws_sagemaker_domain.go (hunk 7 above)
Comment (DrFaust92) [reply aws/resource_aws_sagemaker_domain.go:175]: Ill add this but it requires 2 resource that are WIP (image and image version) and this field is optional

Comment (DrFaust92) [reply aws/resource_aws_sagemaker_domain.go:175]: done

File: website/docs/r/sagemaker_domain.html.markdown (hunk 5 above)
Comment (DrFaust92) [reply website/docs/r/sagemaker_domain.html.markdown:82]: done

**Comment by DrFaust92** [comment]
Rebased + some minor fix in the image resource after using it in tests

File: aws/resource_aws_sagemaker_domain_test.go [hunk 13]
```diff
@@ -0,0 +1,778 @@
...
+  domain_name = %[1]q
+  auth_mode   = "IAM"
+  vpc_id      = aws_vpc.test.id
+  subnet_ids  = [aws_subnet.test.id]
+
+  default_user_settings {
+    execution_role = aws_iam_role.test.arn
+
+    sharing_settings {
+      domain_output_option = "Allowed"
```
Comment (ewbankkit) [inline aws/resource_aws_sagemaker_domain_test.go:681]:
```
=== CONT  TestAccAWSSagemakerDomain_sharingSettings
    resource_aws_sagemaker_domain_test.go:194: Step 1/2 error: Error running pre-apply refresh: 2021/01/09 15:47:27 [DEBUG] Using modified User-Agent: Terraform/0.12.26 HashiCorp-terraform-exec/0.12.0
        
        Error: Unsupported argument
        
          on config040635969/terraform_plugin_test.tf line 57, in resource "aws_sagemaker_domain" "test":
          57:       domain_output_option = "Allowed"
        
        An argument named "domain_output_option" is not expected here.
        
--- FAIL: TestAccAWSSagemakerDomain_sharingSettings (0.57s)
```

**Comment by ewbankkit** [comment]
```console
$ make testacc TEST=./aws/ TESTARGS='-run=TestAccAWSSagemakerDomain_' ACCTEST_PARALLELISM=1
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -count 1 -parallel 1 -run=TestAccAWSSagemakerDomain_ -timeout 120m
=== RUN   TestAccAWSSagemakerDomain_basic
=== PAUSE TestAccAWSSagemakerDomain_basic
=== RUN   TestAccAWSSagemakerDomain_tags
=== PAUSE TestAccAWSSagemakerDomain_tags
=== RUN   TestAccAWSSagemakerDomain_securityGroup
=== PAUSE TestAccAWSSagemakerDomain_securityGroup
=== RUN   TestAccAWSSagemakerDomain_sharingSettings
=== PAUSE TestAccAWSSagemakerDomain_sharingSettings
=== RUN   TestAccAWSSagemakerDomain_tensorboardAppSettings
=== PAUSE TestAccAWSSagemakerDomain_tensorboardAppSettings
=== RUN   TestAccAWSSagemakerDomain_tensorboardAppSettingsWithImage
=== PAUSE TestAccAWSSagemakerDomain_tensorboardAppSettingsWithImage
=== RUN   TestAccAWSSagemakerDomain_kernelGatewayAppSettings
=== PAUSE TestAccAWSSagemakerDomain_kernelGatewayAppSettings
=== RUN   TestAccAWSSagemakerDomain_jupyterServerAppSettings
=== PAUSE TestAccAWSSagemakerDomain_jupyterServerAppSettings
=== RUN   TestAccAWSSagemakerDomain_disappears
=== PAUSE TestAccAWSSagemakerDomain_disappears
=== CONT  TestAccAWSSagemakerDomain_basic
--- PASS: TestAccAWSSagemakerDomain_basic (199.52s)
=== CONT  TestAccAWSSagemakerDomain_tensorboardAppSettingsWithImage
--- PASS: TestAccAWSSagemakerDomain_tensorboardAppSettingsWithImage (299.01s)
=== CONT  TestAccAWSSagemakerDomain_disappears
--- PASS: TestAccAWSSagemakerDomain_disappears (170.70s)
=== CONT  TestAccAWSSagemakerDomain_jupyterServerAppSettings
--- PASS: TestAccAWSSagemakerDomain_jupyterServerAppSettings (259.91s)
=== CONT  TestAccAWSSagemakerDomain_kernelGatewayAppSettings
--- PASS: TestAccAWSSagemakerDomain_kernelGatewayAppSettings (166.19s)
=== CONT  TestAccAWSSagemakerDomain_sharingSettings
    resource_aws_sagemaker_domain_test.go:194: Step 1/2 error: Error running pre-apply refresh: 2021/01/09 15:47:27 [DEBUG] Using modified User-Agent: Terraform/0.12.26 HashiCorp-terraform-exec/0.12.0
        
        Error: Unsupported argument
        
          on config040635969/terraform_plugin_test.tf line 57, in resource "aws_sagemaker_domain" "test":
          57:       domain_output_option = "Allowed"
        
        An argument named "domain_output_option" is not expected here.
        
--- FAIL: TestAccAWSSagemakerDomain_sharingSettings (0.57s)
=== CONT  TestAccAWSSagemakerDomain_tensorboardAppSettings
--- PASS: TestAccAWSSagemakerDomain_tensorboardAppSettings (166.36s)
=== CONT  TestAccAWSSagemakerDomain_securityGroup
--- PASS: TestAccAWSSagemakerDomain_securityGroup (187.38s)
=== CONT  TestAccAWSSagemakerDomain_tags
--- PASS: TestAccAWSSagemakerDomain_tags (202.22s)
FAIL
FAIL	github.com/terraform-providers/terraform-provider-aws/aws	1651.957s
FAIL
GNUmakefile:27: recipe for target 'testacc' failed
make: *** [testacc] Error 1
```

File: aws/resource_aws_sagemaker_domain_test.go (hunk 13 above)
Comment (DrFaust92) [reply aws/resource_aws_sagemaker_domain_test.go:681]: Bizzare... tests passed for me i might have messed up some rebase

Comment (DrFaust92) [reply aws/resource_aws_sagemaker_domain_test.go]: Fixed! Thanks for catching this

**Comment by DrFaust92** [comment]
Fixed issue with non existent/misused argument and added support for kms_key_id argument.

**Review by ewbankkit — APPROVED** [review]
LGTM.

```console
$ make testacc TEST=./aws/ TESTARGS='-run=TestAccAWSSagemakerDomain_' ACCTEST_PARALLELISM=1
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -count 1 -parallel 1 -run=TestAccAWSSagemakerDomain_ -timeout 120m
=== RUN   TestAccAWSSagemakerDomain_basic
=== PAUSE TestAccAWSSagemakerDomain_basic
=== RUN   TestAccAWSSagemakerDomain_kms
=== PAUSE TestAccAWSSagemakerDomain_kms
=== RUN   TestAccAWSSagemakerDomain_tags
=== PAUSE TestAccAWSSagemakerDomain_tags
=== RUN   TestAccAWSSagemakerDomain_securityGroup
=== PAUSE TestAccAWSSagemakerDomain_securityGroup
=== RUN   TestAccAWSSagemakerDomain_sharingSettings
=== PAUSE TestAccAWSSagemakerDomain_sharingSettings
=== RUN   TestAccAWSSagemakerDomain_tensorboardAppSettings
=== PAUSE TestAccAWSSagemakerDomain_tensorboardAppSettings
=== RUN   TestAccAWSSagemakerDomain_tensorboardAppSettingsWithImage
=== PAUSE TestAccAWSSagemakerDomain_tensorboardAppSettingsWithImage
=== RUN   TestAccAWSSagemakerDomain_kernelGatewayAppSettings
=== PAUSE TestAccAWSSagemakerDomain_kernelGatewayAppSettings
=== RUN   TestAccAWSSagemakerDomain_jupyterServerAppSettings
=== PAUSE TestAccAWSSagemakerDomain_jupyterServerAppSettings
=== RUN   TestAccAWSSagemakerDomain_disappears
=== PAUSE TestAccAWSSagemakerDomain_disappears
=== CONT  TestAccAWSSagemakerDomain_basic
--- PASS: TestAccAWSSagemakerDomain_basic (180.67s)
=== CONT  TestAccAWSSagemakerDomain_tensorboardAppSettingsWithImage
--- PASS: TestAccAWSSagemakerDomain_tensorboardAppSettingsWithImage (242.06s)
=== CONT  TestAccAWSSagemakerDomain_disappears
--- PASS: TestAccAWSSagemakerDomain_disappears (164.54s)
=== CONT  TestAccAWSSagemakerDomain_jupyterServerAppSettings
--- PASS: TestAccAWSSagemakerDomain_jupyterServerAppSettings (215.70s)
=== CONT  TestAccAWSSagemakerDomain_kernelGatewayAppSettings
--- PASS: TestAccAWSSagemakerDomain_kernelGatewayAppSettings (197.69s)
=== CONT  TestAccAWSSagemakerDomain_securityGroup
--- PASS: TestAccAWSSagemakerDomain_securityGroup (199.20s)
=== CONT  TestAccAWSSagemakerDomain_tensorboardAppSettings
--- PASS: TestAccAWSSagemakerDomain_tensorboardAppSettings (263.39s)
=== CONT  TestAccAWSSagemakerDomain_sharingSettings
--- PASS: TestAccAWSSagemakerDomain_sharingSettings (170.31s)
=== CONT  TestAccAWSSagemakerDomain_tags
--- PASS: TestAccAWSSagemakerDomain_tags (214.93s)
=== CONT  TestAccAWSSagemakerDomain_kms
--- PASS: TestAccAWSSagemakerDomain_kms (167.24s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	2015.822s
```

**Comment by breathingdust** [comment]
LGTM :rocket: Thanks @DrFaust92!

Verified Acceptance Tests in Commercial (us-west-2)

```console
make testacc TEST=./aws/ TESTARGS='-run=TestAccAWSSagemakerDomain_' ACCTEST_PARALLELISM=1
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -count 1 -parallel 1 -run=TestAccAWSSagemakerDomain_ -timeout 120m
=== RUN   TestAccAWSSagemakerDomain_basic
=== PAUSE TestAccAWSSagemakerDomain_basic
=== RUN   TestAccAWSSagemakerDomain_kms
=== PAUSE TestAccAWSSagemakerDomain_kms
=== RUN   TestAccAWSSagemakerDomain_tags
=== PAUSE TestAccAWSSagemakerDomain_tags
=== RUN   TestAccAWSSagemakerDomain_securityGroup
=== PAUSE TestAccAWSSagemakerDomain_securityGroup
=== RUN   TestAccAWSSagemakerDomain_sharingSettings
=== PAUSE TestAccAWSSagemakerDomain_sharingSettings
=== RUN   TestAccAWSSagemakerDomain_tensorboardAppSettings
=== PAUSE TestAccAWSSagemakerDomain_tensorboardAppSettings
=== RUN   TestAccAWSSagemakerDomain_tensorboardAppSettingsWithImage
=== PAUSE TestAccAWSSagemakerDomain_tensorboardAppSettingsWithImage
=== RUN   TestAccAWSSagemakerDomain_kernelGatewayAppSettings
=== PAUSE TestAccAWSSagemakerDomain_kernelGatewayAppSettings
=== RUN   TestAccAWSSagemakerDomain_jupyterServerAppSettings
=== PAUSE TestAccAWSSagemakerDomain_jupyterServerAppSettings
=== RUN   TestAccAWSSagemakerDomain_disappears
=== PAUSE TestAccAWSSagemakerDomain_disappears
=== CONT  TestAccAWSSagemakerDomain_basic
--- PASS: TestAccAWSSagemakerDomain_basic (258.25s)
=== CONT  TestAccAWSSagemakerDomain_tensorboardAppSettingsWithImage
--- PASS: TestAccAWSSagemakerDomain_tensorboardAppSettingsWithImage (309.42s)
=== CONT  TestAccAWSSagemakerDomain_disappears
--- PASS: TestAccAWSSagemakerDomain_disappears (153.13s)
=== CONT  TestAccAWSSagemakerDomain_jupyterServerAppSettings
--- PASS: TestAccAWSSagemakerDomain_jupyterServerAppSettings (163.91s)
=== CONT  TestAccAWSSagemakerDomain_kernelGatewayAppSettings
--- PASS: TestAccAWSSagemakerDomain_kernelGatewayAppSettings (349.73s)
=== CONT  TestAccAWSSagemakerDomain_securityGroup
--- PASS: TestAccAWSSagemakerDomain_securityGroup (325.53s)
=== CONT  TestAccAWSSagemakerDomain_tensorboardAppSettings
--- PASS: TestAccAWSSagemakerDomain_tensorboardAppSettings (246.46s)
=== CONT  TestAccAWSSagemakerDomain_sharingSettings
--- PASS: TestAccAWSSagemakerDomain_sharingSettings (164.01s)
=== CONT  TestAccAWSSagemakerDomain_tags
--- PASS: TestAccAWSSagemakerDomain_tags (199.56s)
=== CONT  TestAccAWSSagemakerDomain_kms
--- PASS: TestAccAWSSagemakerDomain_kms (164.55s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	2336.269s
```

Verified Acceptance Tests in GovCloud (us-gov-west-1)

SageMaker domains are part of SageMaker studio and as such are not supported in GovCloud. I will create an issue to track the work required to skip these tests when run in that region and link the issue here.

```console
=== CONT  TestAccAWSSagemakerDomain_basic
    resource_aws_sagemaker_domain_test.go:76: Step 1/2 error: Error running apply: 2021/01/13 15:48:21 [DEBUG] Using modified User-Agent: Terraform/0.12.29 HashiCorp-terraform-exec/0.12.0

        Error: error creating SageMaker domain: UnknownOperationException: The requested operation is not supported in the called region.
        	status code: 400, request id: 28cc1e92-93f0-4f19-b6a9-d93ba7ab99f3


--- FAIL: TestAccAWSSagemakerDomain_basic (7.81s)
```

**Pull request merged**